### ISSUE TITLE
Compiling with Mac OS

### ISSUE THREAD

**Issue by Gabriel-El-Nino** [jfm92/microByte_firmware#10]
Hello! 

I am planning to retake this project and expend it as part of my job as an hardware engineer. However, I use the mac OS environment and I have been struggling for a few days already to get it to build using idf.py build. Is it possible to get some help on it / a tutorial so anyone with mac os can compile the firmware on their own? 

Friendly,
Gabriel

**Comment by jfm92** [comment]
Hii @Gabriel-El-Nino !

I didn't use macOS, but I think it should be more or less the same as on Linux. 

You can find here the set-up guide for the toolchain 4.1 (This is the version that use this project) for macOS 
https://docs.espressif.com/projects/esp-idf/en/v4.1/get-started/macos-setup.html

I suggest to start with a basic program like blink to  know if it's building an flashing properly

**Comment by Gabriel-El-Nino** [comment]
Hey @jfm92,

Thank you so much for reaching out back! 

It took me a long while to figure it out, but ended up making it work! 

Initially I tried building your code using the idf.py build command, but kept running into so so so many issues. Eventually I stop trying this route. I can't get pass the "Failed to resolve component 'lvgl'." CMake Error. I have no idea what line is looking for 'lvgl' and no idea where 'lvgl' is. I saw you had to rename a lot components from lvgl to LVGL in one of your previous commit, but can't seem to find that one lvgl thats causing issue. 

I tried following your instruction but initially got stuck at the make -j4 command. And eventually found out I had the newer version of the esp-idf toolchain, which didn't have the make/project.mk in the IDF_PATH. Reverting the esp-idf toolchain to v4.1 resolved the issue and I was able to make -j4 and also flash it. 

Now, I am working to get everything working (screen, button and sound) on a breadboard first and then going to make my own pcb for it! I'm building the retro gaming console so it has a gameboy-ish format. 

I'd be more than happy to share back all my work and help contribute to your project. If you want to get in touch feel free to reach out! Really love your work :D

Cheers
